src/realm/descriptor.hpp

```diff
@@ -326,6 +334,11 @@ class Descriptor : public std::enable_shared_from_this<Descriptor> {
     //@}
 
     //@{
+    /// Returns the contained spec.
+    Spec* get_spec() const;
+    //@}
```

Comment:
However, if you choose to elevate `Spec` to public status, you it seems to me like you must delegate constness from the `Descriptor` to the `Spec` class in this functions.